src/ksm/client/discovery.go

```diff
@@ -118,12 +117,20 @@ func (sd *discoverer) Discover(timeout time.Duration) (client.HTTPClient, error)
 	}
 
 	sd.logger.Debugf("KSM client created with endpoint=%v and nodeIP=%v", endpoint, nodeIP)
-	return newKSMClient(timeout, nodeIP, endpoint, sd.logger, sd.k8sClient), nil
+
+	KSMClient, err := newKSMClient(timeout, nodeIP, endpoint, sd.logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create KSM http client: %w", err)
```

Comment:
[NIT] I prefer to wrap errors with different sentences to understand where they are coming from. In this case we have the same sentence in 3 funcitons

Reply:
(I am not even sure it is a best practice to do so)